src/bun.js/bindings/ZigGlobalObject.cpp

```diff
@@ -1761,6 +1764,12 @@ void GlobalObject::finishCreation(VM& vm)
     Base::finishCreation(vm);
     ASSERT(inherits(info()));
 
+    // Change prototype from null to object for synthetic modules.
+    m_moduleNamespaceObjectStructure.initLater(
+        [] (const Initializer<Structure>& init) {
+            init.set(JSModuleNamespaceObject::createStructure(init.vm, init.owner, init.owner->objectPrototype()));
```

Comment:
I'm not sure exactly of the long-term implications of this change

It is technically against the spec

I don't see a way around it

We should probably do it

Reply:
I agree. And if we really need to go with spec, we can change webkit to make synthetic module record use another structure instead of `m_moduleNamespaceObjectStructure`.